app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java

```diff
@@ -1268,6 +1274,28 @@ public void recenterMap(fr.free.nrw.commons.location.LatLng curLatLng) {
         mapBox.animateCamera(CameraUpdateFactory.newCameraPosition(position), 1000);
     }
 
+    @Override
+    public void showLocationOffDialog() {
+        // This creates a dialog box that prompts the user to enable location
+        DialogUtil
+            .showAlertDialog(getActivity(), getString(R.string.ask_to_turn_location_on), getString(R.string.nearby_needs_location),
+                getString(R.string.yes), getString(R.string.no),  this::openLocationSettings, null);
+    }
+
+    @Override
+    public void openLocationSettings() {
+        // This method opens the location settings of the device along with a followup toast.
+        Intent intent = new Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS);
+        PackageManager packageManager = getActivity().getPackageManager();
+
+        if (intent.resolveActivity(packageManager)!= null) {
+            startActivity(intent);
+            Toast.makeText(getContext(), R.string.set_location_mode_high_accuracy, Toast.LENGTH_LONG).show();
```

Comment:
Looks like you've forgot to update this after updating the strings file.

```suggestion
            Toast.makeText(getContext(), R.string.recommend_high_accuracy_mode, Toast.LENGTH_LONG).show();
```

Reply:
> Looks like you've forgot to update this after updating the strings file.

Ah! I am sorry. I was just too excited last night to get this PR merged since it's practically my first PR to any open source project.

Reply:
It's totally understandable. I still distinctly remember how it felt when I had my first ever open source patch accepted was accepted. It was like flying the air ✈️. Ironically [the patch](https://bugs.python.org/issue28181) was a change to the antigravity python module 😄

Anyways, congratulation on your first PR! Well done.

Reply:
>Ironically the patch was a change to the antigravity python module smile

XKCD, wow. A man of culture :clap: